runtime/src/accounts_background_service.rs

```diff
@@ -169,26 +170,52 @@ impl SnapshotRequestHandler {
                     shrink_time.stop();
                 }
 
-                // Generate an accounts package
+                let block_height = snapshot_root_bank.block_height();
+                let snapshot_type = if block_height
+                    % self.snapshot_config.full_snapshot_archive_interval_slots
+                    == 0
+                {
+                    *last_full_snapshot_slot = Some(snapshot_root_bank.slot());
+                    Some(SnapshotType::FullSnapshot)
+                } else if block_height
+                    % self
+                        .snapshot_config
+                        .incremental_snapshot_archive_interval_slots
+                    == 0
+                    && last_full_snapshot_slot.is_some()
+                {
+                    Some(SnapshotType::IncrementalSnapshot(
+                        last_full_snapshot_slot.unwrap(),
+                    ))
+                } else {
+                    None
+                };
+
+                // Snapshot the bank and send over an accounts package
                 let mut snapshot_time = Measure::start("snapshot_time");
-                let r = snapshot_utils::snapshot_bank(
+                let result = snapshot_utils::snapshot_bank(
                     &snapshot_root_bank,
                     status_cache_slot_deltas,
                     &self.accounts_package_sender,
                     &self.snapshot_config.bank_snapshots_dir,
                     &self.snapshot_config.snapshot_archives_dir,
                     self.snapshot_config.snapshot_version,
-                    &self.snapshot_config.archive_format,
+                    self.snapshot_config.archive_format,
                     hash_for_testing,
+                    snapshot_type,
                 );
-                if r.is_err() {
+                snapshot_time.stop();
+
+                // if there was an error while snapshotting the bank, it was due to the
+                // channel, and thus the node is coming down.  Can log it here and continue on.
+                if result.is_err() {
                     warn!(
-                        "Error generating snapshot for bank: {}, err: {:?}",
+                        "Error taking bank snapshot. slot: {}, snapshot type: {:?}, err: {:?}",
                         snapshot_root_bank.slot(),
-                        r
+                        snapshot_type,
+                        result
```

Comment:
Should we also just return here from AccountsBackgroundService and shut the node down?

 Otherwise reasoning what happens if this fails for a full snapshot and then somehow works for the next incremental snapshot is kind of annoying. We would have an incremental snapshot with a base slot that we don't have a full snapshot for xD

Reply:
Handled offline.